pkgs/applications/networking/cluster/terraform-providers/teleport/default.nix

```diff
@@ -0,0 +1,27 @@
+{ lib, fetchFromGitHub, buildGoModule }:
+buildGoModule rec {
+  pname = "terraform-provider-teleport";
+  version = "7.1.3";
+
+  src = fetchFromGitHub {
+    owner = "gravitational";
+    repo = "teleport-plugins";
+    rev = "v${version}";
+    sha256 = "0bmib3ia8d46ylgdcwpyq9424rfhb6y9yccinn2pwfn8q0jysfjg";
+  };
+  vendorSha256 = null;
+
+  sourceRoot = "source/terraform";
```

Comment:
This is the sole reason why it uses a custom derivation instead of the standardized `providers.json`: the provider lives under a subdirectory in the repo. I am not opposed to introducing a `sourceRoot` field to `providers.json` either, but this seemed like a less invasive option.